src/overlays/actors/ovl_Bg_Hidan_Dalm/z_bg_hidan_dalm.c

```diff
@@ -21,17 +29,175 @@ const ActorInit Bg_Hidan_Dalm_InitVars = {
+        { { { 305.0f, 0.0f, -300.0f }, { 305.0f, 600.0f, 300.0f }, { 305.0f, 0.0f, 300.0f } } },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000040, 0x00, 0x00 }, 0x00, 0x79, 0x00 },
+        { { { -305.0f, 0.0f, -300.0f }, { -305.0f, 600.0f, 300.0f }, { -305.0f, 600.0f, -300.0f } } },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000040, 0x00, 0x00 }, 0x00, 0x79, 0x00 },
+        { { { -305.0f, 0.0f, -300.0f }, { -305.0f, 0.0f, 300.0f }, { -305.0f, 600.0f, 300.0f } } },
+    },
+};
+
+#define sTrisInit D_80886590
+ColliderTrisInit sTrisInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x00, 0x20, COLSHAPE_TRIS },
+    4,
+    sTrisItemInit,
+};
+
+#define sInitChain D_808865A0
+InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, 65336, ICHAIN_STOP),
+};
+
+Vec3f D_808865A8 = { 0, 0, 0 };
+
+#ifdef NON_MATCHING
+// regalloc
+void BgHidanDalm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanDalm* this = THIS;
+    s32 switchFlag;
+    u32 dynaUnk;
+
+    dynaUnk = 0;
+    Actor_ProcessInitChain(thisx, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_UNK);
+    DynaPolyInfo_Alloc(&D_0600DA10, &dynaUnk);
+    // = ... & 0xFFFFFFFFFFFFFFFF; moves regalloc issues
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, dynaUnk);
+    Collider_InitTris(globalCtx, &this->unk_16C);
+    Collider_SetTris(globalCtx, &this->unk_16C, thisx, &sTrisInit, this->unk_18C);
+
+    switchFlag = thisx->params >> 8; /*
+    // this instead of the line above solves half the regalloc issues
+    switchFlag = thisx->params;
+    switchFlag = switchFlag >> 8 & 0xFFFFFFFFFFFFFFFF;
+    //*/
+    this->switchFlag = switchFlag; // can use thisx->params >> 8 without changes
+    thisx->params &= 0xFF;
+    // can use thisx->params >> 8 , seems to improve regalloc, but it changes codegen
+    if (Flags_GetSwitch(globalCtx, switchFlag & 0xFF)) {
+        Actor_Kill(thisx);
+    } else {
+        this->actionFunc = &BgHidanDalm_WaitInteraction;
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Dalm/BgHidanDalm_Init.s")
+#endif
+
+void BgHidanDalm_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanDalm* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Dalm/BgHidanDalm_Destroy.s")
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    Collider_DestroyTris(globalCtx, &this->unk_16C);
+}
+
+void BgHidanDalm_WaitInteraction(BgHidanDalm* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->unk_16C.base.acFlags & 2) && !func_8008E988(globalCtx) &&
+        (player->swordAnimation == 22 || player->swordAnimation == 23)) {
+        this->unk_16C.base.acFlags &= ~(1 << 1);
+        if (this->unk_16C.list[0].body.bumperFlags & 2 || this->unk_16C.list[1].body.bumperFlags & 2) {
```

Comment:
everywhere else in the codebase we have bit flags as hex. They will definitely becomes defines that show the shift value in the future, but they should probably stay hex for the time being 
(lots of these comments that im adding are going to be relevant for en_box too)